app/src/main/java/com/example/sharingang/MessageFragment.kt

```diff
@@ -0,0 +1,225 @@
+     * one value from the other.
+     */
+    private class LinkedPair<A>(private val fst: A, private val snd: A) {
+        /**
+         * Retrieve the pair's other element
+         *
+         * @param x The pair's element whose partner we want to get
+         * @return The pair's partner (null if x is not part of the pair)
+         */
+        fun otherOf(x: A): A? {
+            return if (x == fst) snd else if (x == snd) fst else null
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentMessageBinding.inflate(inflater, container, false)
+        binding.history.layoutManager = LinearLayoutManager(requireContext())
+        (binding.history.layoutManager as LinearLayoutManager).stackFromEnd = true
+        messagesLiveData.observe(viewLifecycleOwner, { newList ->
+            (binding.history.adapter as MessageAdapter).submitList(newList)
+        })
+        listChats = mutableListOf()
+        partnerId = args.partnerId
+        partnerUsername = args.partnerUsername
+        partnerProfilePic = args.partnerProfilePictureUrl
+        currentUserId = currentUserProvider.getCurrentUserId()!!
+        setupFields()
+        setupSendButton()
+        setupUI()
+        setupMessageRefresh()
+        return binding.root
+    }
+
+    /**
+     * Setup the views for the profile picture and the username
+     * of the chat partner on the top of the fragment.
+     */
+    private fun setupFields() {
+        Glide.with(this).load(partnerProfilePic).into(binding.chatPartnerPic)
+        binding.chatPartnerUsername.text = partnerUsername
+    }
+
+    /**
+     * Setup the View and action for the button used for sending a message.
+     */
+    private fun setupSendButton() {
+        binding.btnSend.isEnabled = false
+        binding.messageEditText.addTextChangedListener {
+            binding.btnSend.isEnabled = !it.isNullOrBlank()
+        }
+        binding.btnSend.setOnClickListener {
+            val message: String = binding.messageEditText.text.toString()
+            sendMessage(currentUserId, partnerId, message)
+        }
+    }
+
+    /**
+     * Retrieve the user's document from the database based on its id.
+     *
+     * @param userId the id of the user we want to get the document of
+     * @return the document of the user
+     */
+    private fun getUserDocument(userId: String): DocumentReference {
+        val usersCollection = firebaseFirestore.collection(getString(R.string.users))
+        return usersCollection.document(userId)
+    }
+
+    /**
+     * Puts a message into the database and clears the message text field.
+     *
+     * @param from the sender
+     * @param to the receiver
+     * @param message the message to send
+     */
+    private fun sendMessage(from: String, to: String, message: String) {
+        listChats.clear()
```

Comment:
No, it is not. I have been using it because there was a bug with the list so I just wanted to clear it wherever I could, but now that issue is resolved and `listChats.clear()` is not needed 😄